src/NodaTime/TimeInterval.cs

```diff
@@ -0,0 +1,284 @@
+        /// values.
+        /// </summary>
+        /// <remarks>
+        /// If the start is null, the TimeInterval is deemed to stretch to the start of time. If the end is null,
+        /// the TimeInterval is deemed to stretch to the end of time.
+        /// </remarks>
+        /// <param name="start">The start <see cref="LocalTime"/> or null.</param>
+        /// <param name="end">The end <see cref="LocalTime"/> or null.</param>
+        public TimeInterval(LocalTime? start, LocalTime? end)
+        {
+            this.start = start;
+            this.end = end;
+            if (start.HasValue && end.HasValue)
+                IsEndInNextDay = start > end;
+            else
+                IsEndInNextDay = false;
+        }
+
+        /// <summary>
+        /// Gets the start LocalTime - the inclusive lower bound of the TimeInterval.
+        /// </summary>
+        /// <value>The start <see cref="LocalTime"/>.</value>
+        /// <exception cref="InvalidOperationException">The TimeInterval extends to the start of time.</exception>
+        /// <seealso cref="HasStart"/>
+        public LocalTime Start
+        {
+            get
+            {
+                Preconditions.CheckState(start.HasValue, "TimeInterval extends to start of time");
+                return start.GetValueOrDefault();
+            }
+        }
+
+        /// <summary>
+        /// Returns <c>true</c> if this TimeInterval has a fixed start point, or <c>false</c> if it
+        /// extends to the start of time.
+        /// </summary>
+        /// <value><c>true</c> if this TimeInterval has a fixed start point, or <c>false</c> if it
+        /// extends to the start of time.</value>
+        public bool HasStart => start != null;
+
+        /// <summary>
+        /// Gets the end LocalTime - the exclusive upper bound of the TimeInterval.
+        /// </summary>
+        /// <value>The end <see cref="LocalTime"/>.</value>
+        /// <exception cref="InvalidOperationException">The TimeInterval extends to the end of time.</exception>
+        /// <seealso cref="HasEnd"/>
+        public LocalTime End
+        {
+            get
+            {
+                Preconditions.CheckState(end.HasValue, "TimeInterval extends to end of time");
+                return end.GetValueOrDefault();
+            }
+        }
+
+        /// <summary>
+        /// Returns <c>true</c> if this TimeInterval has a fixed end point, or <c>false</c> if it
+        /// extends to the end of time.
+        /// </summary>
+        /// <value><c>true</c> if this TimeInterval has a fixed end point, or <c>false</c> if it
+        /// extends to the end of time.</value>
+        public bool HasEnd => end.HasValue;
+
+
+        /// <summary>
+        /// Returns <c>true</c> if Start of TimeInterval is greater than End (for intervals like 23:00/1:00)
+        /// </summary>
+        public bool IsEndInNextDay { get; }
+
+        /// <summary>
```

Comment:
I'm not sure this worth having as a property. Periods are *usually* around calendrical calculations.
What we should probably have is just the Duration property, and then Duration.ToPeriod (which I'm slightly surprised we don't already have). Then at least the functionality is simple to achieve.

We can always add this property later if necessary.

Reply:
Ok, should I delete the code or just make Period method internal?

Reply:
I'd just remove it entirely.

Reply:
Removed.